var/spack/repos/builtin/packages/openssl/package.py

```diff
@@ -33,13 +33,17 @@ class Openssl(Package):
        Layer Security (TLS) and Secure Sockets Layer (SSL) protocols.
        It is also a general-purpose cryptography library."""
     homepage = "http://www.openssl.org"
-    url = "ftp://openssl.org/source/openssl-1.0.1h.tar.gz"
+    url      = "ftp://openssl.org/source/openssl-1.0.1h.tar.gz"
+    list_url = "https://www.openssl.org/source/old/"
```

Comment:
ftp does work, you just need to get rid of the `www.` so that it matches how the url looks.